src/clinicaltrialgov.ts

```diff
@@ -65,10 +83,40 @@ export function findNCTNumber(study: ResearchStudy): string | null {
 export function parseClinicalTrialXML(fileContents: string): Promise<ClinicalStudy> {
   const parser = new xml2js.Parser();
   return parser.parseStringPromise(fileContents).then((result) => {
-    return (result as TrialBackup).clinical_study;
+    if (isTrialBackup(result)) {
+      return result.clinical_study;
+    } else {
+      throw new Error('Unable to parse trial as valid clinical study XML');
+    }
   });
 }
 
+const CLINICAL_STATUS_MAP = new Map<StatusEnum, ResearchStudyStatus>([
+  ['Active, not recruiting', 'closed-to-accrual'],
+  ['Completed', 'completed'],
+  // FIXME: This does not appear to have a proper mapping
+  ['Enrolling by invitation', 'active'],
+  ['Not yet recruiting', 'approved'],
+  ['Recruiting', 'active'],
+  ['Suspended', 'temporarily-closed-to-accrual'],
+  ['Terminated', 'administratively-completed'],
+  ['Withdrawn', 'withdrawn'],
+  // FIXME: Guessing on this mapping
+  ['Available', 'active'],
+  ['No longer available', 'closed-to-accrual'],
+  ['Temporarily not available', 'temporarily-closed-to-accrual'],
+  // FIXME: what?
+  ['Approved for marketing', 'active'],
```

Comment:
This one is similar to Available above, it probably should map to completed since it is available for the general public. Not sure how often we will run into this one though.